From 9849c09c6a34ec364399c8e0918a073678e150b2 Mon Sep 17 00:00:00 2001
From: Maksim Rakitin <mrakitin@bnl.gov>
Date: Thu, 2 Dec 2021 19:38:38 -0500
Subject: [PATCH] PKG: patch package_data

---
 setup.py | 22 +++++++++++++++++++---
 1 file changed, 19 insertions(+), 3 deletions(-)

diff --git a/setup.py b/setup.py
index 99af0f50..77744c16 100644
--- a/setup.py
+++ b/setup.py
@@ -115,11 +115,27 @@ packages = ['larch', 'larch.bin']
 for pname in find_packages('larch'):
     packages.append('larch.%s' % pname)
 
+import platform
 
-package_data = ['icons/*', 'xray/*.dat', 'xray/*.db', 'xrd/*.db',
-                'bin/darwin64/*', 'bin/linux64/*', 'bin/win64/*',
-                'bin/win32/*']
+# See https://docs.python.org/3/library/platform.html for details.
+mach = platform.machine()
+system = platform.system()
+arch = platform.architecture()  # returns something like `('64bit', 'ELF')`
+
+if system == "Windows":
+    libsrc = "win"
+else:
+    libsrc = system.lower()
+
+if mach.startswith("arm"):
+    nbits = "arm"
+else:
+    nbits = arch[0].replace("bit", "")
 
+plat = libsrc + nbits
+
+package_data = ['icons/*', 'xray/*.dat', 'xray/*.db', 'xrd/*.db',
+                f'bin/{plat}/*']
 
 if INSTALL:
     # before install:  remove historical cruft, including old plugins
-- 
2.34.1

